api/v1alpha1/authorization_types.go

```diff
@@ -0,0 +1,47 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+// Authorization defines the authorization configuration.
+type Authorization struct {
+	// Rules contains all the authorization rules.
```

Comment:
what happens when there are many rules, but none of them match, is the default action to `Deny` ?
if so we should add a comment on it

Reply:
if we have any allow rule and it does not match, then the default action is Deny

Reply:
thanks, what about the case of multiple rules matching for the same request - Rule 1's action is `allow`, and Rule2's action is `deny`, how is the decision computed here ?

Reply:
lets assume that we have case:

- allow cidr 1.2.3.0/24
- deny cidr 1.2.3.123/32

if the client ip is 1.2.3.123 it will be denied, otherwise all other ips from that 1.2.3.0/24 block will be allowed.

Reply:
thanks for explaining this, and what about 
```
deny cidr 1.2.3.0/24
allow cidr 1.2.3.123/32
```

will `1.2.3.123` get allowed or denied ?

from a users perspective, they'd like to know which of these below are true
* first match or
* best match or
* if any match has a deny action , then deny

Reply:
that does not work (or depends on the filter order?). In that case 1.2.3.123 will be denied. I have not tested but it might be because of the order of allow and deny rbac filters. 

Earlier my rbac filters was in order: allow, deny. So if someone wants that working then it might be possible if filters are in order: deny, allow. 

I am thinking how this is actually going to be implemented, if we have following:

allow cidr 8.0.0.0/8
deny cidr 8.1.0.0/16
allow cidr 8.1.1.1/32

how this is translated to filters? Should we keep the order and make own rbac filter from each clientSelector item (and rules inside single clientSelector is just ANDed)? Then it might be possible to make these kind of things. (not sure I have not tested).